python/cudf/cudf/tests/test_replace.py

```diff
@@ -359,6 +359,11 @@ def test_series_multiple_times_with_nulls():
 
     for i in range(3):
         got = sr.replace([1, 2, 3], None)
-        # This print is required as it will fail only if there is print
-        print(got)
         assert_eq(expected, got)
+        # The following series will acquire a chunk of memory and update with
```

Comment:
You are making a lot of assumptions that depend on underlying memory allocator / reuse implementation details. Every "will" in the below comment is actually a "may". Even then it was just luck that this test reproduced the bug.  I think it would be good to refer to the issue number in the comment in case anyone needs to do code archaeology on this in the future.

Reply:
ok, I will update the comment.